### PR TITLE
Move our OnOffSwitches to Switch component from @patternfly/react-core

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
@garrett  @martinpitt I am not sure where I should import the patternfly.css file
I have now imported in the end of the pf-overwrites files, but definitely it doesn't belong there.
Other than that, I tried the switch component, which works brilliant, other than a small issue, which I was not able to fix.
There is some outline around the component, when it's focused, that I can't get rid of, and I am not sure what it's caused from.

**Comment by marusak** [comment]
Needs to rebase to master since #12367 changed tests names

**Comment by martinpitt** [comment]
This doesn't build right now, as it conflicts with the recently reorganized  Services page. That still has one button left:

    pkg/systemd/service-details.jsx:import { OnOffSwitch } from "cockpit-components-onoff.jsx";

File: pkg/networkmanager/firewall.jsx [hunk 1]
```diff
@@ -921,9 +921,9 @@ export class Firewall extends React.Component {
                 </ol>
                 <h1>
                     {_("Firewall")}
-                    <OnOffSwitch state={enabled}
-                                 disabled={!!this.state.pendingTarget}
-                                 onChange={this.onSwitchChanged} />
+                    <Switch id='firewall-onoff-switch' isChecked={enabled}
+                        isDisabled={this.state.pendingTarget}
```
Comment (martinpitt) [inline pkg/networkmanager/firewall.jsx]: Let's please keep the `!!` here. pendingTarget can be `null`, which is not a valid boolean.

**Comment by martinpitt** [comment]
This seems to interfere rather heavily with our custom PF3 CSS, or the PF4 CSS is not included correctly:

switch

I. e. the checkmark is misplaced and too small, the switch text is too small, and the blue focus ring is ugly (as you mentioned).

The `patternfly-font-paths` test failure is right -- we can't ship a patternfly.css with references like "src:url(../../node_modules/@patternfly/patternfly/assets/pficon/pficon.eot?#iefix)", they won't exist in an installation. This goes with your larger question of where to include the CSS. I need to look at this more closely,  presumably as a first step we need to start shipping new fonts in cockpit-ws and then start massaging these paths in tools/patternfly.sed like we do for OpenSans.

For now I just force-pushed a rebase to master (bots changes), and porting the OnOffSwitch element on the Services page, so that this at least builds again.

**Comment by martinpitt** [comment]
> I am not sure where I should import the patternfly.css file

Do we need this at all? ISTM  that PF4 components should, and do, just import their own CSS in their React component. If I simply revert your changes to src/base1/patternfly-overrides.less, then the switches look a bit better, still work, and our PF3 patternfly.css does not get broken either:

switch-noimport

I. e. in the future we should  pretend that our PF3 `<link href="../base1/patternfly.css"</link>` does not exist at all.

Of course this may be too simplistic for components which rely on additional fonts and such (unless they embed them in their CSS), but it seems at least for the Switch it may not be required yet? node_modules/@patternfly/patternfly/components/Switch/ doesn't seem to include anything external.

**Comment by KKoukiou** [comment]
@martinpitt I don't know why the jpg files can be used in our current code without these loaders.

Without the two loaders, I get in this PR:
```
ERROR in ./node_modules/@patternfly/react-styles/css/components/AboutModalBox/about-modal-box.css
Module build failed (from ./node_modules/extract-text-webpack-plugin/dist/loader.js):
ModuleParseError: Module parse failed: Unexpected character '�' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```

**Comment by garrett** [comment]
The Red Hat fonts are imported anyway, so if they are referenced, then they are used. And if the page mixes in PF3, then PF4 widgets should "know" to use the Red Hat fonts instead of Overpass. And even if this is "broken", the fonts are defined on our pages at a top level, so the style cascade should cover it.

TL;DR: Fonts should just work. (Perhaps not in PF4-only pages. But perhaps even then?)

**Comment by KKoukiou** [comment]
@garrett do you think you can get rid of that surrounding blue line?
I can fix the tests in the meantime.

**Comment by KKoukiou** [comment]
@martinpitt 

Current master:
[kkoukiou@sequoia cockpit]$ du -sh dist/
51M	dist/

This branch
[kkoukiou@sequoia cockpit]$ du -sh dist/
52M	dist/

So it doesn't make such big difference I guess.

**Comment by KKoukiou** [comment]
Blocked by https://github.com/patternfly/patternfly-react/issues/2606

**Comment by garrett** [comment]
@KKoukiou Ugh... 1 megabyte is actually quite huge for a single page to transfer over networks and for a browser to interpret and process.

Can you do a `du -sk` instead, please, so it doesn't round to the nearest meg?

(FWIW: 51/52 megs is gargantuan — and Cockpit needs to be put on a diet — but at least it's not all downloaded at the same time in one page. And, ideally, it should be mostly cached.)
